### ISSUE TITLE
Found what appears to be bug

### ISSUE THREAD

**Issue by visualperception** [GuardRex/vscode-status-bar-tasks#9]
my tasks.json code is:

if args conatains a vscode variable such as ${relativeFile}, then it doesn't get translated when passing it to a browser. I get following in my browser url:

`http://localhost/$%7BrelativeFile%7D`

running task directly from vscode it does get translated.

my code is:

```
{
    // See https://go.microsoft.com/fwlink/?LinkId=733558
    // for the documentation about the tasks.json format
    "version": "0.1.0",
    "command": "cmd",
    "isShellCommand": true,
    "showOutput": "always",
    "args": ["/C"],
    "tasks": [
        {
            "taskName": "iexplore",
            "suppressTaskName": true,
            "args" : ["iexplore.exe localhost/${relativeFile}"],
            "showOutput": "always"
        }
             ]
}
```

thanks

**Comment by guardrex** [comment]
The extension only supports ...
- ${workspaceRoot} the path of the folder opened in VS Code
- ${file} the current opened file
- ${fileBasename} the current opened file's basename
- ${fileDirname} the current opened file's directory
- ${fileExtname} the current opened file's extension
- ${env.} environment variable (e.g., ${env.MY_ENV_VAR} will be replaced by the value of MY_ENV_VAR on the system or an empty string if the environment variable is not set).

... it's not really a "bug," per se. I think `${relativeFile}` was one that the VS Code team added after I did my work on that part of the extension. I'll take a look in the coming days (but it might be two weeks! I'm getting slammed with work right now.) and see if it's something I can roll into the next update.

**Comment by guardrex** [comment]
@visualperception Let me do a head-check with you on this just to make sure ...

`${workspaceRoot}` = *c:\\_APP\\folder1\\folder2\\*
Currently opened file = *c:\\_APP\\folder1\\folder2\\folder3\\somefile.txt*
`${relativeFile}` = *folder3\\somefile.txt* 👈 Is that correct?

**Comment by visualperception** [comment]
${relativeFile}  does NOT have a leading / so you are correct on that

in my tasks json file I now have since the implementation of ${workspaceRootFolderName} 
`{
    // See https://go.microsoft.com/fwlink/?LinkId=733558
    // for the documentation about the tasks.json format
    "version": "0.1.0",
    "command": "cmd",
    "isShellCommand": true,
    "showOutput": "always",
    "args": ["/C"],
    "tasks": [
        {
            "taskName": "iexplore",
            "suppressTaskName": true,
            "args" : ["iexplore","localhost/${workspaceRootFolderName}/${relativeFile}"],
            "showOutput": "always"
        }
             ]
}
`
as you can see I have to put the / between ${workspaceRootFolderName} and ${relativeFile} and this works for me and passes the correct url to iexplore which runs fine on my local host and browser.

${workspaceRootFolderName} has NO slashes in it at all and from your example would contain just folder2.

localhost/${workspaceRootFolderName}/${relativeFile}

resolves to:

localhost/folder2/folder3/somfile.txt

**Comment by guardrex** [comment]
Yes, I do have `${workspaceRootFolderName}` with no slashes ... just the last folder name from the workspace root, so I think I'm good there.

I just wanted to make sure that my definition of "relative" matches the common notion here as I show above.

You also raise an interesting point here wrt `\` (backslashes) and `/` (forward slashes). On non-Windows OS's, I take it that VSC is dealing with forward-slash paths? If so, I have a little more work to do! This thingy was built on Windows ... mostly for my own use. It became much more popular (relatively speaking ... >2K installs) than I ever expected. I'll see if I can fix that up in this round of updates.

**Comment by guardrex** [comment]
:+1: Yes. I have it working now.

I just need to deal with the `/` forward slashes situation for `/` forward slash platforms.

This will make the next update, and I'm mostly just going to be waiting on them to get back to me on another issue/feature before I roll out the update. The version will be v0.1.1, and I hope to get it out later tonight or tomorrow depending on how fast they get back to me.

Temporarily blocked by: https://github.com/Microsoft/vscode/issues/17566

**Comment by guardrex** [comment]
Yep. Looking good. When this version hits, you can test the var replacements with this task ...
```json
{
    "taskName": "test variable replacements",
    "suppressTaskName": true,
    "args" : ["echo", "file: ${file} fileBasename: ${fileBasename} relativeFile: ${relativeFile} fileDirname: ${fileDirname} fileExtname: ${fileExtname} workspaceRoot: ${workspaceRoot} fileBasenameNoExtension: ${fileBasenameNoExtension} workspaceRootFolderName: ${workspaceRootFolderName}"],
    "isTestCommand": true,
    "showOutput": "always"
}
```
I'm just waiting now on one of the VSC devs to get back to me and let me know what to do about the file schema for `tasks.json` so that the `"showInStatusBar": false` property will work.

**Comment by guardrex** [comment]
This is fixed with latest version in the gallery (v.0.1.2). Thanks again for pointing out the problem.
